lib/model/autocomplete.dart

```diff
@@ -163,6 +183,48 @@ final RegExp _emojiIntentRegex = (() {
     + r')$');
 })();
 
+final RegExp _channelLinkIntentRegex = () {
+  // What's likely to come just before #channel syntax: the start of the string,
+  // whitespace, or punctuation. Letters are unlikely; in that case a GitHub-
+  // style "zulip/zulip-flutter#124" link might be intended (as on CZO where
+  // there's a custom linkifier for that).
+  //
+  // By punctuation, we mean *some* punctuation, like "(". We make "#" and "@"
+  // exceptions, to support typing "##channel" for the channel query "#channel",
+  // and typing "@#user" for the mention query "#user", because in 2025-11
+  // channel and user name words can start with "#". (They can also contain "#"
+  // anywhere else in the name; we don't handle that specially.)
```

Comment:
Hmmm. What's the behavior if the user tries to search for such a name, then? It'll fail, right? They want the channel `yay #1 go team`; they type `#yay #1`, and we treat it as a channel query `1`, not `yay #1`.

I think it's not important to handle channel (or user) names like `#1 winner`, nor like `yay #1 go team`. Particularly so if we're still only handling a subset of the names that in principle are valid. It's fine to support them if it doesn't make things appreciably more complex, but not worth any significant complexity.

Reply:
Well, that's a small enough piece of logic that I guess it's pretty harmless to include. So maybe just edit the comment where it says "we don't handle that specially", so that it answers the question I asked above.